### PR TITLE
Publisher read only mode improvements 

### PR DESCRIPTION
**Pull request by hasuniea**
https://docs.google.com/document/d/1VNlv2HEB7EN_kk_57GmA65KtqO5vTu2eYjjS83VFf9E/edit?usp=sharing

### CODE REVIEW

File: portals/publisher/source/src/app/components/Base/Header/avatar/Avatar.jsx [hunk 1]
```diff
@@ -90,6 +91,13 @@ class Avatar extends Component {
         if (user.name.endsWith('@carbon.super') && count <= 1) {
             username = user.name.replace('@carbon.super', '');
         }
+        let usr;
+        const readOnlyUser = AuthManager.isReadOnlyUser();
+        if (readOnlyUser) {
+            usr = username + ' (read-only)';
```
Comment (tmkasun) [inline portals/publisher/source/src/app/components/Base/Header/avatar/Avatar.jsx:97]: we have to make this i18n compatible, please use <FormattedMessage/> component

Comment (tmkasun) [reply portals/publisher/source/src/app/components/Base/Header/avatar/Avatar.jsx:97]: image
Shall we display it like this ?

[source](https://excalidraw.com/#json=vBja0Bv2iL73S_yhzHTPw,aNNm0MlbkZg7t9c36lLcqA)

File: portals/publisher/source/src/app/data/AuthManager.js [hunk 2]
```diff
@@ -164,6 +164,25 @@ class AuthManager {
...
+     * @returns {Boolean} isReadOnlyUser
+     * @memberof AuthManager
+     */
+    static isReadOnlyUser() {
+        if (AuthManager.getUser() === null) {
+            return false;
+        } else {
+            const arrayLength = AuthManager.getUser().scopes.length;
+            if (arrayLength === 2) {
+                return AuthManager.getUser().scopes.includes('apim:api_view', 'openid');
```
Comment (tmkasun) [inline portals/publisher/source/src/app/data/AuthManager.js:180]: `Array.includes` does not accept multiple values, We have to use either lodash intercept or a simplified version of it using pure JS

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

**Review by tmkasun — CHANGES_REQUESTED** [review]

File: portals/publisher/source/src/app/components/Base/Header/avatar/Avatar.jsx (hunk 1 above)
Comment (hasuniea) [reply portals/publisher/source/src/app/components/Base/Header/avatar/Avatar.jsx:97]: fixed

File: portals/publisher/source/src/app/data/AuthManager.js [hunk 3]
```diff
@@ -19,6 +19,7 @@
 import qs from 'qs';
 import CONSTS from 'AppData/Constants';
 import Configurations from 'Config';
+import _ from 'lodash';
```
Comment (tmkasun) [inline portals/publisher/source/src/app/data/AuthManager.js:22]: Shall we use the [lodash.intersectionby](https://www.npmjs.com/package/lodash.intersectionby) package similar to [what we have here](https://github.com/wso2/apim-apps/blob/main/portals/publisher/package.json#L53-L54)

**Review by tmkasun — APPROVED** [review]

**Pull request merged**